zson/resolver.go

```diff
@@ -0,0 +1,329 @@
+package zson
```

Comment:
Nit: Call this file `typecontext.go`.

Reply:
Or `context.go` if you rename `TypeContext` to `Context`!

Reply:
as discussed will change zson.TypeContext to zson.Context and name this file context.go